tests/draft2019-09/additionalItems.json

```diff
@@ -35,23 +35,41 @@
     {
         "description": "array of items with no additionalItems",
         "schema": {
-            "items": [{}, {}, {}],
+            "items": [
+                {"type": "integer"},
+                {"type": "string"}
+            ],
             "additionalItems": false
         },
         "tests": [
             {
-                "description": "fewer number of items present",
-                "data": [ 1, 2 ],
+                "description": "empty array",
+                "data": [ ],
                 "valid": true
             },
             {
-                "description": "equal number of items present",
-                "data": [ 1, 2, 3 ],
+                "description": "fewer number of items present, valid",
+                "data": [ 1 ],
                 "valid": true
             },
+            {
+                "description": "fewer number of items present, invalid",
+                "data": [ "a" ],
+                "valid": false
+            },
+            {
+                "description": "equal number of items present, valid",
+                "data": [ 1, "a" ],
+                "valid": true
+            },
+            {
+                "description": "equal number of items present, invalid",
+                "data": [ "a", 1 ],
+                "valid": false
+            },
             {
                 "description": "additional items are not permitted",
-                "data": [ 1, 2, 3, 4 ],
+                "data": [ 1, 2, 3 ],
```

Comment:
This test previously failed for the reason it says, but now it may fail for totally unrelated reasons, namely that the second element doesn't match the schema for it.

(In general we have to be quite careful about *mutating* existing tests -- appending new ones to it is safe, but yeah here this seems worrying)

Which is the real new test here that isn't covered? Like even the "fewer number of items present" seem like they should be covered already via combination of this test and the tests covering items. Is there an actual bug that prompted this or is this just to flesh out more cases?

Reply:
> now it may fail for totally unrelated reasons

ah, good point.

Yes, I am trying to add coverage for implementation bugs I was finding, that weren't exposed by existing test cases.

I shall attempt to create more concise tests that more clearly expose each issue in isolation.